flake.nix

```diff
@@ -53,67 +51,112 @@
         });
 
       # The default package for 'nix build'. This makes sense if the
       # flake provides only one package or there is a clear "main"
       # package.
-      defaultPackage = forAllSystems (system: self.packages.${system}.hello);
+      defaultPackage = forAllSystems (system: self.packages.${system}.irma-server);
 
-      # A NixOS module, if applicable (e.g. if the package provides a system service).
-      nixosModules.hello =
-        { pkgs, ... }:
-        {
-          nixpkgs.overlays = [ self.overlay ];
+      
+      devShell = forAllSystems (system:
+        let
+          pkgs = import nixpkgs {
+            inherit system;
+            overlays = [ self.overlay ];
+          };
+
+        in
+        pkgs.mkShell {
+
+
+          buildInputs = with pkgs; [
+            go
+            postgresql
+            mailhog
+          ];
+
+          shellHook = ''
+              export PGHOST=$HOME/postgres
+              export PGDATA=$PGHOST/data
+              export PGDATABASE=postgres
+              export PGLOG=$PGHOST/postgres.log
+            
+              if pg_ctl status 
+              then 
+              pg_ctl stop
+              fi
+
+              if [ ! -d $PGDATA ]; then
+                initdb -U postgres --auth=trust --no-locale --encoding=UTF8
+              fi
+              
+              pg_ctl -D $PGDATA -l $PGLOG -o "-k /tmp" -o "-F -p 5432" start
+              
+              if  pg_ctl status
+              then
+              createuser -h localhost -s postgres
+
+              psql -U postgres -h localhost -c 'CREATE database test'
+              psql -U postgres -h localhost -c "CREATE USER testuser with encrypted password 'testpassword'"
+              psql -U postgres -h localhost -c 'grant all privileges on database test to testuser'
+              fi
+            '';
 
-          environment.systemPackages = [ pkgs.hello ];
 
-          #systemd.services = { ... };
-        };
+
+        });
 
       # Tests run by 'nix flake check' and by Hydra.
       checks = forAllSystems (system: {
-        inherit (self.packages.${system}) hello;
+         inherit (self.packages.${system}) irma-server;
+         test =
+         with nixpkgsFor.${system};
+        stdenv.mkDerivation {
+            name = "irma-server-test-${version}";
 
-        # Additional tests, if applicable.
-        test =
-          with nixpkgsFor.${system};
-          stdenv.mkDerivation {
-            name = "hello-test-${version}";
+            buildInputs = [ irma-server go postgresql mailhog ];
 
-            buildInputs = [ hello ];
+            src = irma-server-src;
+          
+            vendorSha256 =  "sha256-9JZKl5hm3qtbYNwPFyDSzndhw6/Qr2DN/CY3eSNDMxU=";
 
             unpackPhase = "true";
 
-            buildPhase = ''
-              echo 'running some integration tests'
-              [[ $(hello) = 'Hello, world!' ]]
+            postInstall = ''
+              export PGHOST=$HOME/postgres
```

Comment:
How about trying to run those as vm test?
Grafana has a good example of a test with a service that has a db
https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/grafana.nix
Running all those stateful changes in a bash script feels like it depends on implementation details. Implementing it in with nixos test should be less brittle when upgrades happen.

Reply:
will try to implement the vm test